pkg/controller/elasticsearch/migration/migrate_data.go

```diff
@@ -76,12 +88,45 @@ func IsMigratingData(ctx context.Context, shardLister esclient.ShardLister, podN
 	return nodeIsMigratingData(podName, shards, excludedNodes), nil
 }
 
+// AllocationExcludeAnnotationName returns the allocation exclude value stored in an annotation.
+// May be empty if not set.
+func allocationExcludeFromAnnotation(es esv1.Elasticsearch) string {
```

Comment:
I thought this kind of problem was detected by the linter ?

Reply:
It looks like it is disabled by default: https://github.com/golangci/golangci-lint/issues/843

Reply:
> also not sure if I would have put this map lookup in a function in the first place (that needs a comment) given that it is used exactly once

That's true. I wanted to separate the high-level logic (`MigrateData`) from the storage mechanism (annotations get/set), but indeed it feels a bit artificial here since getting from a map is pretty simple. No strong opinion, if I have to choose I would keep the current code though.